### PR TITLE
Reorganize and polish the contribution section

### PR DESCRIPTION
**Pull request by pbrezina**
This is major update to the existing contents.

### CODE REVIEW

**Comment by pbrezina** [comment]
Added architecture document.

File: src/contrib/introduction.rst [hunk 1]
```diff
@@ -0,0 +1,101 @@
+Introduction to the SSSD development
+####################################
+
+Thank you for your interest in contributing to the SSSD project. The development
+is happening on the `SSSD GitHub project`_ using the `git`_ version control
+system and `GitHub workflow`_. SSSD is a large complex project with many
+dependencies. Feel free to reach us on our `community channels`_ to get the help
+you require.
```
Comment (justin-stephenson) [inline src/contrib/introduction.rst:8]: s/get the help you require/ask for help/

File: src/contrib/introduction.rst [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+.. seealso::
+
+    Check out also other `GitHub Guides <https://guides.github.com>`_ if you are
+    not familiar with GitHub.
+
+Contribution Quick Start
+========================
+
+There are plenty of guides laying around to get you started with git and GitHub
```
Comment (justin-stephenson) [inline src/contrib/introduction.rst:23]: s/laying around/available/

File: src/contrib/introduction.rst [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+workflow if you are not yet familiar with these, therefore we don't go into
+details here. As an example, we suggest reading the `Git Handbook`_ on GitHub.
+The following steps describe the process in a nut shell to provide you a basic
+template:
+
+.. _Git Handbook: https://guides.github.com/introduction/git-handbook
+
+#. Get familiry with the :doc:`coding-style`
```
Comment (justin-stephenson) [inline src/contrib/introduction.rst:31]: s/familiry/familiar/

File: src/contrib/building-sssd.rst [hunk 4]
```diff
@@ -1,90 +1,90 @@
 Building SSSD
-=============
+#############
 
-Starting with SSSD 1.10 beta, we now include a set of helper aliases and
-environment variables in SSSD to simplify building development versions of
-SSSD on Fedora. To take advantage of them, use the following command:
+SSSD is using `autoconf`_ and `automake`_ build systems. If you are familiar
+with these tools, you already know what to do. If you are not, these are the
+commands you need to execute.
```
Comment (justin-stephenson) [inline src/contrib/building-sssd.rst:6]: I would change this to the following, or something similar.
```
SSSD is using `autoconf`_ and `automake`_ build systems. If you are not familiar with these tools, below are the
commands you need to execute.
```

File: src/contrib/building-sssd.rst [hunk 5]
```diff
@@ -1,90 +1,90 @@
...
+    [sssd]$ autoreconf -if
+    [sssd]$ ./configure --enable-nsslibdir=/lib64 --enable-pammoddir=/lib64/security --enable-silent-rules
+    [sssd]$ make
 
-- Install necessary tools:
+The ``autoreconf`` command will create a ``configure`` script. This script
+allows you to modify the resulting build with several flags, it also makes sure
+that all required dependencies are available. SSSD requires lots of build
+dependencies that helps us integrate with the identity and authentication
+systems. Chances are that you don't have all the dependencies installed. In such
```
Comment (justin-stephenson) [inline src/contrib/building-sssd.rst:21]: s/Chances are that you don't have all the dependencies installed/Your system may not have all necessary dependencies installed/

File: src/contrib/building-sssd.rst [hunk 6]
```diff
@@ -1,90 +1,90 @@
...
 
-Sidenote: By default autotools install prefix is "/usr/local". Default location
-for "nsslibdir" and "pammoddir" will be "/usr/local/lib64".
-32bit machines will search for SSSD libraries by default in "/usr/local/lib"
-which will result in broken NSS and PAM linking with SSSD. Advice here is to
-double check if SSSD install location will be visible for system linker.
+.. code-block:: console
 
-- Build RPM packages out of fresh build if needed
+    [sssd]$ reconfig && chmake
```
Comment (justin-stephenson) [inline src/contrib/building-sssd.rst:83]: Should mention that `chmake` will run unit tests?

**Comment by justin-stephenson** [comment]
This looks great, especially the architecture page. Nice work.

**Comment by pbrezina** [comment]
Thanks. I'm going to push it.

**Pull request merged**